azurelinuxagent/common/utils/shellutil.py

```diff
@@ -120,14 +120,14 @@ class CommandError(Exception):
     Exception raised by run_command when the command returns an error
     """
     @staticmethod
-    def _get_message(command, returncode):
+    def _get_message(command, return_code, stderr):
         command_name = command[0] if isinstance(command, list) and len(command) > 0 else command
-        return "'{0}' failed: {1}".format(command_name, returncode)
+        return "'{0}' failed: {1} ({2})".format(command_name, return_code, stderr.rstrip())
```

Comment:
debugging failures with just the error code in the exception message can be hard; added stderr

Reply:
Can there be a case where stderr is None? If it is `stderr.rstrip()` would throw

Reply:
no, it'd be an empty string

Reply:
Awesome (Y)